packages/admin/admin/src/mui/menu/Menu.styles.ts

```diff
@@ -1,58 +1,66 @@
+import { CSSObject, Drawer as MuiDrawer, DrawerProps, Theme } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import { createStyles, StyledComponent } from "@mui/styles";
+import { MUIStyledCommonProps } from "@mui/system";
+import * as React from "react";
 
-import { MenuProps } from "./Menu";
+import { MasterLayoutContext } from "../MasterLayoutContext";
+import { DEFAULT_DRAWER_WIDTH, DEFAULT_DRAWER_WIDTH_COLLAPSED, MenuProps } from "./Menu";
 
-export type MenuClassKey = "drawer" | "permanent" | "temporary" | "open" | "closed";
+const openedMixin = (theme: Theme, drawerWidth?: number): CSSObject => ({
+    width: drawerWidth ?? DEFAULT_DRAWER_WIDTH,
+    transition: theme.transitions.create("width", {
+        easing: theme.transitions.easing.sharp,
+        duration: theme.transitions.duration.enteringScreen,
+    }),
+    overflowX: "hidden",
+});
+const closedMixin = (theme: Theme, drawerVariant: DrawerProps["variant"], drawerWidth?: number, drawerWidthCollapsed?: number): CSSObject => ({
+    width: drawerVariant === "temporary" ? drawerWidth ?? DEFAULT_DRAWER_WIDTH : drawerWidthCollapsed ?? DEFAULT_DRAWER_WIDTH_COLLAPSED,
+    transition: theme.transitions.create("width", {
+        easing: theme.transitions.easing.sharp,
+        duration: theme.transitions.duration.leavingScreen,
+    }),
+    overflowX: "hidden",
+});
+export const Drawer: StyledComponent<DrawerProps & MUIStyledCommonProps<Theme> & Pick<MenuProps, "drawerWidth" | "drawerWidthCollapsed">> = styled(
+    MuiDrawer,
+    { shouldForwardProp: (prop) => prop !== "drawerWidth" && prop !== "drawerWidthCollapsed" },
+)<DrawerProps & MUIStyledCommonProps<Theme> & Pick<MenuProps, "drawerWidth" | "drawerWidthCollapsed">>(
+    ({ theme, open, variant, drawerWidth, drawerWidthCollapsed }) => {
+        const { headerHeight } = React.useContext(MasterLayoutContext);
 
-export const styles = (theme: Theme) =>
-    createStyles<MenuClassKey, MenuProps>({
-        drawer: {
-            "& [class*='MuiDrawer-paper']": {
-                backgroundColor: "#fff",
-            },
-            "& [class*='MuiPaper-root']": {
-                flexGrow: 1,
-                overflowX: "hidden",
+        return {
+            ...(variant === "permanent" && {
+                backgroundColor: theme.palette.common.white,
+                flexShrink: 0,
+                whiteSpace: "nowrap",
+                boxSizing: "border-box",
+                ...(open ? openedMixin(theme, drawerWidth) : closedMixin(theme, variant, drawerWidth, drawerWidthCollapsed)),
+            }),
+            "& .MuiDrawer-paper": {
+                backgroundColor: theme.palette.common.white,
+                ...(variant === "permanent" && {
+                    top: headerHeight,
+                    height: `calc(100% - ${headerHeight}px)`,
+                }),
+                ...(open ? openedMixin(theme, drawerWidth) : closedMixin(theme, variant, drawerWidth, drawerWidthCollapsed)),
```

Comment:
Why is the animation-mixin set on both, the permanent drawer and the drawer's paper?
This would mean, two nested elements have the same width-animation when the permanent drawer is shown, right? Is that really necessary?

Reply:
https://github.com/vivid-planet/comet/assets/56400587/055f5d8e-2f68-41d2-bd80-587073a5bd1d

Reply:
The first version is like it is, the second one with the first mixin commented out and the third with the second mixin commented out

Reply:
Ah ok I get it now, the drawer is what pushes the page-content away and the paper is positioned with `fixed` so it needs an additional animation. Makes sense 👍🏼